doc/dev/design/NAT-address-discovery.rst

```diff
@@ -0,0 +1,143 @@
+    -  Disadvantages: More standardization effort? How do we solve authentication?
+
+3.  Extend SCMP with a new ERROR message: "invalid source address for first hop pkt", similar to error 33.
+    The router can verify that for first hop packets, the IP src address/port matches the SCION src address/port.
+    If not, it returns an error. The actual source address would need to be attached somewhere, unless we decide
+    to change the payload so it contains the IP header of the offending packet (and the IP header should contain the
+    NATed IP/port).
+
+    -  Advantages: One less dependency on an external library and protocol
+    -  Disadvantages: Conceptually a bit of a hack. More standardization effort? How do we solve authentication?
+
+
+Rationale
+=========
+[A discussion of alternate approaches and the trade-offs, advantages, and disadvantages of the specified approach.]
+
+The main reasons for integrating the functionality with the BR are:
+
+-  Reliability: The border router is almost guaranteed to see the correct IP/port on the NAT, especially when using the
+   same port for STUN traffic and routing traffic. All other approaches rely on the leniency of the NAT to use the same
+   port even if the STUN server and border router have different ports or even IPs.
+-  Time to rollout: changing the border routers should be much easier and faster than getting router vendors or NAT
+   vendors to implement SCION compatibility or to get rid of NATs completely in home networks.
+   A short time until rollout seems important because people are already running into this problem.
+
+Alternatives:
+
+-  The SRC address/port is updated by the border router to reflect what the border router sees as source address.
+   Problem:
+   -  Complicates cryptographically protecting the header if the header must be modifiable by border routers
+   -  A spoofed IP causes the border router to route traffic to an unsuspecting target
+-  The SRC address/port is updated by the NAT. This is similar to having the border router update tSRC IP/port.
+   - Complicates cryptographically protecting the header if the header must be modifiable by border routers
+-  Use STUN servers. This is a possibility, but adds setup complexity and may not work in all cases. Every subnet
+   of an AS that has a border router must also have a STUN server. Moreover, if the STUN server uses a different IP
+   (or port) than the border router, then the NAT may decide to use a different port when connecting to it, i.e. the
+   STUN server may not see the same IP/port tuple on the NAT that the border router sees. Disadvantages:
+
+   -  This approach may be be problematic with sensitive NATs.
+   -  We need to somehow standardize the STUN IP/port and/or communicate it to endhosts, e.g. via the topo file or
+      its successor.
+-  Remove all NATs and use IPv6 instead. This is technically possible but unlikely to happen anytime soon.
```

Comment:
Maybe add something along the lines "especially because scarcity of IPv4 addresses is not the only reason why NATs are deployed".